runtime/entry-names.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines the macro RTNAME(n) which decorates the external name of a runtime
+// library function or object with extra characters so that it
+// (a) is not in the user's name space,
+// (b) doesn't conflict with other libraries, and
+// (c) prevents incompatible versions of the runtime library from linking
+
+#ifndef RTNAME
+#define RTNAME(n) _F18RTa_##n
```

Comment:
We should probably use "flang" rather than "f18" as that's what the compiler will eventually be called.
Perhaps `_FlangA_##n` or `_Flang1_##n`.

Reply:
The `F` is for Fortran and the 18 is for the 2018 standard.  If a later standard introduces a change that isn't backwards compatible, we'll update the number and reset the revision letter.

Reply:
OK, makes sense.

Reply:
I've pushed a change that makes the names longer and adds comments.